glusterd2/volume/fs_utils.go

```diff
@@ -144,3 +154,155 @@ func GetMounts() ([]*Mntent, error) {
+func IsMountExist(b *brick.Brickinfo, iD uuid.UUID, mtab []*Mntent) bool {
+
+	if _, err := os.Lstat(b.Path); err != nil {
+		return false
+	}
+
+	data := make([]byte, 16)
+	sz, err := syscall.Getxattr(b.Path, volumeIDXattrKey, data)
+	if err != nil || sz <= 0 {
+		return false
+	}
+
+	//Check for little or big endian ?
+	if !uuid.Equal(iD, data[:sz]) {
+		return false
+	}
+
+	if mtab == nil {
+		//Skip high level mount validation
+		return true
+	}
+
+	mountData := b.MountInfo
+	mountRoot := strings.TrimSuffix(b.Path, mountData.BrickDir)
+	for _, entry := range mtab {
+		if entry.MntDir == mountRoot {
+			if entry.MntType != mountData.FsType {
+				return false
+			}
+			devicePath, err := os.Readlink(mountData.DevicePath)
+			if err != nil {
+				return false
+			}
+			deviceMapper, err := os.Readlink(entry.FsName)
+			if err != nil {
+				return false
+			}
+			if deviceMapper == devicePath {
+				return true
+			}
+			//No need to continue as mount root already processed
+			break
+		}
+	}
+	return false
+}
+
+//UmountBrickDirectory does an umount of the path
+func UmountBrickDirectory(path string) error {
+	err := syscall.Unmount(path, syscall.MNT_FORCE)
+	return err
+}
+
+//MountBrickDirectory creates the directory strcture for bricks
+func MountBrickDirectory(vol *Volinfo, brickinfo *brick.Brickinfo, mtab []*Mntent) error {
+
+	if !brickinfo.SelfProvision {
+		return nil
+	}
+
+	mountData := brickinfo.MountInfo
+	mountRoot := strings.TrimSuffix(brickinfo.Path, mountData.BrickDir)
+	//Because of abnormal shutdown of the brick, mount point might already be existing
+	if IsMountExist(brickinfo, vol.ID, mtab) {
+		log.WithFields(log.Fields{
+			"mountRoot":  mountRoot,
+			"fsType":     mountData.FsType,
+			"devicePath": mountData.DevicePath,
+		}).Debug("Mount point already exist")
+		return nil
+	}
+
+	if err := os.MkdirAll(mountRoot, os.ModeDir|os.ModePerm); err != nil {
+		log.WithError(err).Error("Failed to create snapshot directory ", brickinfo.String())
+		return err
+	}
+
+	if err := MountDirectory(mountRoot, brickinfo.MountInfo); err != nil {
+		log.WithFields(log.Fields{"error": err.Error(),
+			"brickPath": brickinfo.String(),
+			"mountRoot": mountRoot}).Error("Failed to mount snapshot directory")
+
+		return err
+	}
+
+	if err := unix.Setxattr(brickinfo.Path, volumeIDXattrKey, vol.ID, 0); err != nil {
+		log.WithFields(log.Fields{"error": err.Error(),
+			"brickPath": brickinfo.Path,
+			"xattr":     volumeIDXattrKey}).Error("setxattr failed")
+		return err
+	}
+
+	return nil
+}
+
+//MountDirectory will mount the bricks to the given path
+func MountDirectory(mountPath string, mountData brick.MountInfo) error {
+	err := utils.ExecuteCommandRun("mount", "-o", mountData.MntOpts, mountData.DevicePath, mountPath)
+	// Use syscall.Mount command to mount the bricks
+	return err
+}
+
+//StopBrick terminate the process and umount the brick directory
+func StopBrick(b brick.Brickinfo, logger log.FieldLogger) error {
+	var err error
```

Comment:
TRIVIAL:  IMO this declaration is not required, you can directly use err := b.Terminate.. and err:= UmountBrick.

Also you can directly use ` return UmountBrick(b)`